bin/jsii-release-maven

```diff
@@ -39,151 +41,182 @@ set -eu # we don't want "pipefail" to implement idempotency
 #
 ###
 
+readonly CENTRAL_SERVER="ossrh"
+scriptdir="$(cd $(dirname $0) && pwd)"
+
 error() { echo "❌ $@"; exit 1; }
 
-cd "${1:-"dist/java"}"
+validate_central_parameters() {
+    [[ -z "${MAVEN_STAGING_PROFILE_ID:-}" ]] && error "MAVEN_STAGING_PROFILE_ID is required"
+    return 0
+}
 
-[ -z "${MAVEN_STAGING_PROFILE_ID:-}" ] && error "MAVEN_STAGING_PROFILE_ID is required"
-[ -z "${MAVEN_USERNAME:-}" ] && error "MAVEN_USERNAME is required"
-[ -z "${MAVEN_PASSWORD:-}" ] && error "MAVEN_PASSWORD is required"
+validate_non_central_parameters() {
+    [[ -z "${MAVEN_SERVER_ID:-}" ]] && error "MAVEN_SERVER_ID is required"
+    [[ -z "${MAVEN_REPOSITORY_URL:-}" ]] && error "MAVEN_REPOSITORY_URL is required"
+    return 0
+}
 
-[ -z "${MAVEN_GPG_PRIVATE_KEY_PASSPHRASE:-}" ] && error "MAVEN_GPG_PRIVATE_KEY_PASSPHRASE is required"
-[ -z "${MAVEN_GPG_PRIVATE_KEY_FILE:-}" ] && [ -z "${MAVEN_GPG_PRIVATE_KEY:-}" ] && error "MAVEN_GPG_PRIVATE_KEY_FILE or MAVEN_GPG_PRIVATE_KEY is required"
-[ -n "${MAVEN_GPG_PRIVATE_KEY:-}" ] && [ -n "${MAVEN_GPG_PRIVATE_KEY_FILE:-}" ] && error "Cannot specify both MAVEN_GPG_PRIVATE_KEY and MAVEN_GPG_PRIVATE_KEY_FILE"
+validate_signing_parameters() {
+    [[ -z "${MAVEN_GPG_PRIVATE_KEY_PASSPHRASE:-}" ]] && error "MAVEN_GPG_PRIVATE_KEY_PASSPHRASE is required"
+    [[ -z "${MAVEN_GPG_PRIVATE_KEY_FILE:-}" ]] && [[ -z "${MAVEN_GPG_PRIVATE_KEY:-}" ]] && error "MAVEN_GPG_PRIVATE_KEY_FILE or MAVEN_GPG_PRIVATE_KEY is required"
+    [[ -n "${MAVEN_GPG_PRIVATE_KEY:-}" ]] && [[ -n "${MAVEN_GPG_PRIVATE_KEY_FILE:-}" ]] && error "Cannot specify both MAVEN_GPG_PRIVATE_KEY and MAVEN_GPG_PRIVATE_KEY_FILE"
+    return 0
+}
 
-if [[ -n "${MAVEN_DRYRUN:-}" ]]; then
-    echo "==========================================="
-    echo "            🏜️ DRY-RUN MODE 🏜️"
-    echo "==========================================="
-    mvn="echo mvn"
-    dry_run=true
-else
-    mvn=mvn
-    dry_run=false
-fi
+validate_parameters() {
+    [[ -z "${MAVEN_USERNAME:-}" ]] && error "MAVEN_USERNAME is required"
+    [[ -z "${MAVEN_PASSWORD:-}" ]] && error "MAVEN_PASSWORD is required"
 
-#---------------------------------------------------------------------------------------------------------------------------------------
-# Import private GPG key
-#---------------------------------------------------------------------------------------------------------------------------------------
+    if [[ "${is_central}" == true ]]; then
+        validate_central_parameters
+    else
+        validate_non_central_parameters
+    fi
+
+    if [[ "${is_signed}" == true ]] || [[ "${is_central}" == true ]]; then
+        validate_signing_parameters
+    fi
+}
```

Comment:
That's probably a good idea. I was thinking it would error anyways, but actually it would probably sign packages and then deploy unsigned in the current state.